src/primitives/block.h

```diff
@@ -84,9 +88,10 @@ class CBlockHeader
 
     bool ExtractMinterKey(CKeyID &key) const
     {
-        CPubKey recoveredPubKey{};
-        if (!recoveredPubKey.RecoverCompact(GetHashToSign(), sig)) {
-            return false;
+        if (!recoveredPubKey.IsValid()) {
+            if (!recoveredPubKey.RecoverCompact(GetHashToSign(), sig)) {
+                return false;
+            }
```

Comment:
This function may be not thread safe.

Reply:
It does not need to be. It's used under `cs_main` lock.